snafu-derive/src/shared.rs

```diff
@@ -528,6 +528,20 @@ pub mod display {
 
             let source_field = selector_kind.source_field();
 
+            if field_container.is_transparent {
+                // transparent errors always have a source field
```

Comment:
Should we have a test to exercise what happens if we don't have a source field?

Reply:
It's now tested by the compile-fail test, if that's what you mean. :)